### PR TITLE
Add realname flag command

### PR DESCRIPTION
**Pull request by Siguza**
As per issue #13816.

### CODE REVIEW

File: libr/core/cmd_flag.c [hunk 1]
```diff
@@ -934,6 +935,30 @@ static int cmd_flag(void *data, const char *input) {
 			}
 		}
 		break;
+	case 'N':
+		if (input[1]==' ' && input[2]) {
```
Comment (radare) [inline libr/core/cmd_flag.c]: spaces around =

File: libr/core/cmd_flag.c [hunk 2]
```diff
@@ -934,6 +935,30 @@ static int cmd_flag(void *data, const char *input) {
...
+			char *name, *realname;
```
Comment (radare) [inline libr/core/cmd_flag.c]: assign and define in the same line

File: libr/core/cmd_flag.c [hunk 3]
```diff
@@ -934,6 +935,30 @@ static int cmd_flag(void *data, const char *input) {
...
+			} else {
+				realname = name;
+				item = r_flag_get_i (core->flags, core->offset);
+			}
+			if (item) {
+				r_flag_item_set_realname (item, realname);
+			} else {
+				eprintf ("Cannot find flag (%s)\n", name);
+			}
+		}
```
Comment (radare) [inline libr/core/cmd_flag.c]: show help on else

**Comment by Siguza** [comment]
Done. I had originally copy-pasted the `fr` command code, but I changed that as well now.

File: libr/core/cmd_flag.c [hunk 4]
```diff
@@ -930,7 +930,30 @@ static int cmd_flag(void *data, const char *input) {
...
+			} else {
+				realname = name;
+				item = r_flag_get_i (core->flags, core->offset);
+			}
+			if (item) {
+				r_flag_item_set_realname (item, realname);
+			} else {
+				eprintf ("Usage: fN [[name]] [realname]\n");
 			}
 		}
```
Comment (radare) [inline libr/core/cmd_flag.c]: wait.. if no arguments, we should show the realname of the flag in the current offset imho (or all of them if more than one)

**Comment by radare** [comment]
also can you add some tests after this?

**Comment by Siguza** [comment]
I'd rather keep `fN`, for two reasons:

- If `fN x` changes the current symbol's realname, I'd find it more intuitive for no arguments to show _only the current symbol's realname_, instead of all of them.
- The code case for `fn` looks deeply entangled with `f`, `f*`, `fj` and `fq`:

      case '\0':
      case 'n': // "fn" "fnj"
      case '*': // "f*"
      case 'j': // "fj"
      case 'q': // "fq"
      	if (input[1] == '.' && !input[2]) {
      	// ...

So if it's OK with you, I'd leave `fn` the way it is, but change `fN` once more so that:

- `fN` -> show realname at current offset
- `fN realname` -> change realname at current offset
- `fN name realname` -> change realname of flag "name"

**Comment by radare** [comment]
the help of fn talks about realname. maybe the help is wrong.. we can have your prposal merged, and then (wiuthout having tests) improve the commands to avoid uppercase ones and dupes

**Comment by Siguza** [comment]
Alright, I made the change to `fN`.

And yes, the help of `fn` does talk about realname, but there is one single implementation that handles all of `f`, `fj`, `fq` and `f*`.

**Pull request merged**